### PR TITLE
Fix setIfEmpty transformer - handle unicode strings and null string

### PR DESCRIPTION
**Pull request by erezh31**
## Status
Ready

## Description
Fix setIfEmpty transformer:
1. handle unicode strings 
2. Consider "null" or "None" string as empty
@idovandijk - FYI

### CODE REVIEW

File: Scripts/SetIfEmpty/SetIfEmpty.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ args:
 - auto: PREDEFINED
   default: true
   defaultValue: 'true'
-  description: If set to true, empty strings, arrays and dictionaries are considered
+  description: If set to true, empty strings, the string value "None" and "Null", arrays and dictionaries are considered
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty.yml]:
```suggestion
  description: If set to true, empty strings, arrays and dictionaries and the strings "None" and "Null" are considered
```

**Comment by erezh31** [comment]
@roysagi plz review RN 
thanks

**Comment by erezh31** [comment]
@anara123 a lot of the tests failed due to a change in the common server. please look on that - I don't think this build will pass

**Comment by anara123** [comment]
```
============= Scripts/SetIfEmpty =============
Detected python version: [2.7] for docker image: demisto/python:1.3-alpine
========= Running flake8 ===============
flake8 completed
========= Running mypy on: SetIfEmpty.py ===============
SetIfEmpty.py:20:62: error: "basestring" has no attribute "lower" 
[attr-defined]
    ..._string_null = isinstance(value, basestring) and value.lower() in ["no...
                                                        ^
Found 1 error in 1 file (checked 1 source file)
[FAILED /home/circleci/project/Scripts/SetIfEmpty] Error: Command '['bash', '/home/circleci/project/Tests/scripts/run_mypy.sh', '2.7', 'SetIfEmpty.py']' returned non-zero exit status 1. Output: None
Need a more detailed log? try running with the -v options as so: 
/home/circleci/project/Tests/scripts/pkg_dev_test_tasks.py -d Scripts/SetIfEmpty -v
```

**Comment by anara123** [comment]
@erezh31 
```
Traceback (most recent call last):
  File "./Documentation/common_server_docs.py", line 236, in <module>
    main(sys.argv[1:])
  File "./Documentation/common_server_docs.py", line 220, in main
    'CommonServerPython', 'python')
  File "./Documentation/common_server_docs.py", line 152, in createPyDocumentation
    y = parser.parse_docstring(docstring)
  File "/home/circleci/project/venv/local/lib/python2.7/site-packages/parinx/parser.py", line 254, in parse_docstring
    _check_arguments_dict(arguments_dict)
  File "/home/circleci/project/venv/local/lib/python2.7/site-packages/parinx/parser.py", line 122, in _check_arguments_dict
    'Can not get description for argument %s' % (argument))
parinx.errors.MethodParsingException: Can not get description for argument text
```
it seems to fail to parse `encode_string_results` docs.
I think you should change the indentation

**Comment by bakatzir** [comment]
@David-BMS Is using this PR to address: https://github.com/demisto/etc/issues/21061

**Comment by David-BMS** [comment]
I'm retracting my changes as it doesn't work

**Comment by anara123** [comment]
@erezh31 
```
=================================== FAILURES ===================================
__________________________ test_encode_string_results __________________________

    def test_encode_string_results():
        s = "test"
        assert s == encode_string_results(s)
        s2 = u"בדיקה"
>       assert s2.encode("utf-8") == encode_string_results(s2)
E       AssertionError: assert b'\xd7\x91\xd...7\xa7\xd7\x94' == 'בדיקה'
E         -b'\xd7\x91\xd7\x93\xd7\x99\xd7\xa7\xd7\x94'
E         +'בדיקה'

CommonServerPython_test.py:969: AssertionError
===================== 1 failed, 103 passed in 0.58 seconds =====================
Pytest completed with status code: 1
=========== ERRORS FOUND ===========
pylint/pytest returned errors. pylint: [0], pytest: [1]
====================================
```
the unit test still fails

**Comment by erezh31** [comment]
@anara123 I think now I fixed the UT, only some of the playbooks failed

**Comment by erezh31** [comment]
@anara123 ?

**Comment by anara123** [comment]
@erezh31 our master is red now, when we will stabilize it I will merge the PR

**Pull request merged**